malcolm/modules/ca/parts/cawaveformtablepart.py

```diff
@@ -0,0 +1,50 @@
+from malcolm.core import Part, Widget, PartRegistrar, NumberArrayMeta, TableMeta, DEFAULT_TIMEOUT, Display
+from .. import util
+
+
+class CAWaveformTablePart(Part):
+    """Defines a float64[] `Attribute` that talks to multiple DBR_DOUBLE waveform PV,
+      plus checks HOPR & LOPR on these PVs to determine axis limits"""
+
+    def __init__(self,
+                 name,  # type: util.APartName
+                 description,  # type: util.AMetaDescription
+                 pv_list=(),  # type: util.APvList
+                 name_list=(),  # type: util.ANameList
+                 min_delta=0.05,  # type: util.AMinDelta
+                 timeout=DEFAULT_TIMEOUT,  # type: util.ATimeout
+                 widget=Widget.PLOT,  # type: util.AWidget
+                 group=None,  # type: util.AGroup
+                 config=True,  # type: util.AConfig
+                 display_from_pv=False  # type: util.AGetLimits
+                 ):
+        # type: (...) -> None
+        super(CAWaveformTablePart, self).__init__(name)
+        assert len(pv_list) == len(name_list), "List of PVs must be same length as list of names!"
+        self.display_from_pv = display_from_pv
+        elements = {}
+        for name in name_list:
+            elements[name] = NumberArrayMeta("float64", name, display=Display(), tags=(Widget.TEXTUPDATE.tag(),))
```

Comment:
```suggestion
            elements[name] = NumberArrayMeta("float64", name, tags=[Widget.TEXTUPDATE.tag()])
```
In general where it doesn't matter, I would use a list for readability...